backend/src/test/java/com/redhat/cloud/notifications/db/QueryTest.java

```diff
@@ -78,4 +105,295 @@ public void testSort() {
         assertNull(query.getSortBy());
     }
 
+    /**
+     * Tests that no constraint is violated when the {@link Query#offset}, {@link Query#pageNumber} and
+     * {@link Query#pageSize} fields contain a valid value. Reflection is used to grab the values from the annotations
+     * and to generate a valid value that shouldn't trigger any errors.
+     * @throws NoSuchFieldException if the field doesn't exit in the object.
+     * @throws IllegalAccessException if the field could not be updated via reflection.
+     */
+    @Test
+    public void validMinimumFieldsTest() throws NoSuchFieldException, IllegalAccessException {
+        // Get the annotation's values from the "pageSize" field of the "Query" class.
+        final Field pageSizeField = Query.class.getDeclaredField("pageSize");
+        final Max pageSizeMaxAnnotation = pageSizeField.getAnnotation(Max.class);
+        final Min pageSizeMinAnnotation = pageSizeField.getAnnotation(Min.class);
+        final long maxQueryPageSizeValue = pageSizeMaxAnnotation.value();
+        final long minQueryPageSizeValue = pageSizeMinAnnotation.value();
+
+        // Get the annotation's values from the "pageNumber" field of the "Query" class.
+        final Field pageNumberField = Query.class.getDeclaredField("pageNumber");
+        final Min pageNumberMinAnnotation = pageNumberField.getAnnotation(Min.class);
+        final long minQueryPageNumberValue = pageNumberMinAnnotation.value();
+
+        // Get the annotation's values from the "offset" field of the "Query" class.
+        final Field offsetField = Query.class.getDeclaredField("offset");
+        final Min offsetMinAnnotation = offsetField.getAnnotation(Min.class);
+        final long minQueryOffsetValue = offsetMinAnnotation.value();
+
+        // Get a random number that will be used for the validation. The number must be between the "min" and "max"
+        // values of these annotations, so make sure we generate one that satisfies that constraint.
+        final int minRand = (int) Math.min(minQueryOffsetValue, Math.min(minQueryPageNumberValue, minQueryPageSizeValue));
+        final int validValue = ThreadLocalRandom.current().nextInt(minRand, (int) maxQueryPageSizeValue);
+
+        final Query query = new Query();
+
+        // Set the fields as accessible to be able to set a value.
+        offsetField.setAccessible(true);
+        pageNumberField.setAccessible(true);
+        pageSizeField.setAccessible(true);
```

Comment:
You can instead remove the `private` modifier and use the `default` one as we do with `sort_by` https://github.com/RedHatInsights/notifications-backend/blob/5784bb04af4c9b4060b274ac1140e46643c1f775/backend/src/main/java/com/redhat/cloud/notifications/db/Query.java#L34

Reply:
All right, I wasn't sure if that was okay or not... Modified! :smiley: